src/conf_mode/syslog.py

```diff
@@ -0,0 +1,267 @@
+        {
+          rhost : {
+           'selectors' : generate_selectors(c, 'host ' + rhost + ' facility'), 
+            'proto' : proto
+          } 
+        }
+      )
+
+  ## set system syslog user
+  if c.exists('user'):
+    usrs = c.list_nodes('user')
+    for usr in usrs:
+      config_data['user'].update(
+        {
+          usr : {
+            'selectors' : generate_selectors(c, 'user ' + usr + ' facility')
+          }
+        }
+      )
+
+  return config_data
+
+def generate_selectors(c, config_node):
+## protocols and security are being mapped here
+## for backward compatibility with old configs
+## security and protocol mappings can be removed later
+  if c.is_tag(config_node):
+    nodes = c.list_nodes(config_node)
+    selectors = ""
+    for node in nodes:
+      lvl = c.return_value( config_node + ' ' + node + ' level')
+      if lvl == None:
+        lvl = "err"
+      if lvl == 'all':
+        lvl = '*'
+      if node == 'all' and node != nodes[-1]:
+        selectors += "*." + lvl + ";"
+      elif node == 'all':
+        selectors += "*." + lvl
+      elif node != nodes[-1]:
+        if node == 'protocols':
+          node = 'local7'
+        if node == 'security':
+          node = 'auth'
+        selectors += node + "." + lvl + ";"
+      else:
+        if node == 'protocols':
+          node = 'local7'
+        if node == 'security':
+          node = 'auth'
+        selectors += node + "." + lvl
+    return selectors
+
+def generate(c):
+  tmpl = jinja2.Template(configs, trim_blocks=True)
+  config_text = tmpl.render(c)
+  #print (config_text)
+  with open('/etc/rsyslog.d/vyos-rsyslog.conf', 'w') as f:
+    f.write(config_text)
+
+  ## eventually write for each file its own logrotate file, since size is defined it shouldn't matter
+  tmpl = jinja2.Template(logrotate_configs, trim_blocks=True)
+  config_text = tmpl.render(c)
+  #print (config_text)
+  with open('/etc/logrotate.d/vyos-rsyslog', 'w') as f:
+    f.write(config_text)
+
+def verify(c):
+  if c == None:
+    return None
+
+  fac = ['*','auth','authpriv','cron','daemon','kern','lpr','mail','mark','news','protocols','security',\
+        'syslog','user','uucp','local0','local1','local2','local3','local4','local5','local6','local7']
+  lvl = ['emerg','alert','crit','err','warning','notice','info','debug','*']
+  for conf in c:
+    if c[conf]:
+      for item in c[conf]:
+        for s in c[conf][item]['selectors'].split(";"):
+          f = re.sub("\..*$","",s)
+          if f not in fac:
+            print (c[conf])
+            raise ConfigError('Invalid facility ' + s + ' set in '+ conf + ' ' + item)
+          l = re.sub("^.+\.","",s)
+          if l not in lvl:
+            raise ConfigError('Invalid logging level ' + s + ' set in '+ conf + ' ' + item)
+
+def apply(c):
+  ### vyatta-log.conf is being generated somewhere
+  ### this is just a quick hack to remove the old configfile
+  
+  if os.path.exists('/etc/rsyslog.d/vyatta-log.conf'):
```

Comment:
Are there really situations when it would exist? After image upgrade, /etc/ would be clean.

Reply:
Yes. I did a fresh install, then installed my modified packages and that file did exist. So, the old vyatta-system-cfg generates it somewhere (haven't found yet, but I also didn't look too hard). So updates would be affected. I can also put instead of global a global1 or so as channel name, but I thought I stick with global since it would refer directly to the config section. If both files exist, the one which is loaded first is being the global, which could lead to very confusing runtime behavior.